src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (Object.keys(coords).length === 0) {
+      this.coords = {
+        x: coords.x || 0,
+        y: coords.y || 0,
+      };
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this
+      .name}, Chip version: ${this
+      .chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    if (!this.coords.z) {
+      this.coords.z = coords.z || 0;
+    }
```

Comment:
Additional check is redundant.
```suggestion
    this.coords.z = coords.z || 0;
```